app/controllers/projects_controller.rb

```diff
@@ -24,6 +25,30 @@ def create
 
 private
 
+  def searcher
+    @_search ||= Projects::BulkSearch.new(project, params)
+  end
+
+  def content_items
+    searcher.project_content_items_to_display.map do |content_item|
+      content_item.taxons = taxons_for(content_item)
+      content_item
+    end
+  end
+
+  def taxons_for(content_item)
+    result = content_tags.find { |hsh| hsh['content_id'] == content_item.content_id }
```

Comment:
could potentially use `content_tags.find(->{[]}) { |hsh...` to give it a default value

Reply:
I'm going to replace the publishing-api endpoint so that it returns a hash of content-id => links to make this faster and easier